packages/ui/VirtualList/VirtualListBase.js

```diff
@@ -660,6 +765,60 @@ const VirtualListBaseFactory = (type) => {
 			}
 		}
 
+		// For NewVirtualList
+		adjustVariableGridPosition () {
+			const
+				{dataSize, overhang, spacing, variableGridSizes} = this.props,
+				{firstIndex, numOfItems} = this.state,
+				{itemContainerRef, maxFirstIndex, variableGridPositions} = this,
+				lastIndex = firstIndex + numOfItems - 1,
+				numOfUpperLine = Math.floor(overhang / 2);
+
+			if (itemContainerRef.current) {
+				let index;
+
+				// Cache variable item positions
+				// and adjust variable item DOM element positions
+				for (index = firstIndex; index <= lastIndex; index++) {
+					const childNode = itemContainerRef.current.children[index % numOfItems];
+
+					if (!variableGridPositions[index] && variableGridSizes[index]) {
+						const position = index === 0 ? 0 : this.getVariableGridPosition(index - 1) + spacing;
+
+						variableGridPositions[index] = {position};
+					}
+
+					if (childNode && variableGridPositions[index]) {
+						childNode.style.transform = `translate3d(0, ${variableGridPositions[index].position}px, 0)`;
+					}
+				}
+
+				// Update threshold
+				this.threshold.min = firstIndex === 0 ? -Infinity : this.getVariableGridPosition(firstIndex + numOfUpperLine * 1);
+				this.threshold.max = lastIndex === maxFirstIndex ? Infinity : this.getVariableGridPosition(firstIndex + (numOfUpperLine + 1));
+
+				// Update scroll bounds
+				if (variableGridPositions.filter(Boolean).length === dataSize) { // all item sizes are known
```

Comment:
We already assume that there is no jumping, so we can replace this comparison with `variableGridPositions.length === dataSize` or some other simpler code. But I think it looks better to leave this code as is. 👍

Reply:
With our more discussion, it would be better to fix this code as `variableGridPositions.length === dataSize`.